source/jormungandr/tests/ptref_tests.py

```diff
@@ -778,6 +778,39 @@ def test_pt_objects_with_disable_disruption(self):
         )
         assert len(response['disruptions']) == 1
 
+    def test_networks(self):
+        """test networks API"""
+        response = self.query_region("networks")
+        networks = get_not_null(response, 'networks')
+        assert len(networks) == 4
+        assert networks[0]['id'] == 'network:A'
+        assert networks[1]['id'] == 'network:B'
+        assert networks[2]['id'] == 'network:CaRoule'
+        assert networks[3]['id'] == 'network:freq'
+
+    def test_networks_with_external_code(self):
+        """test networks with external_code parameter"""
+        response = self.query_region("networks?external_code=A")
+        networks = get_not_null(response, 'networks')
+        assert len(networks) == 1
+        assert networks[0]['id'] == 'network:A'
+
+        response = self.query_region("networks?external_code=B")
+        networks = get_not_null(response, 'networks')
+        assert len(networks) == 1
+        assert networks[0]['id'] == 'network:B'
+
+        response = self.query_region("networks?external_code=wrong_code")
+        networks = get_not_null(response, 'networks')
+        assert len(networks) == 4
+
+        # without coverage
+        response, code = self.query_no_assert("v1/networks?external_code=A")
```

Comment:
Could you check that you get a 503 "Not implemented" without external_code param, also?

Reply:
Good idea :bird: